app/services/decidim/erc/crm_authenticable/crm_authenticable_authorization_handler.rb

```diff
@@ -48,6 +54,33 @@ def ws_request_must_succeed
           errors.add(:base, I18n.t("connection_failed", scope: "crm_authenticable.errors")) if response[:error]
         end
 
+        def csv_request_must_succeed
+          return if errors.any?
+
+          file = Rails.application.secrets.csv_users_pre[:path]
+          @response = nil
+
+          CSV.foreach(file) do |row|
+            doc_number = row.first
+
+            if doc_number.present? && doc_number.match?(document_number)
+                @response = {
+                  body: 
+                  [
+                    { 
+                      "display_name": row[1],
+                      "email": row[2],
+                      "phone": row[3],
+                      "custom_21": row[4]
+                    }.transform_keys(&:to_s)
+                  ]
+                }
+            else
+              errors.add(:base, I18n.t("connection_failed", scope: "crm_authenticable.errors"))
```

Comment:
the message should be that the user is not in the census or similar

Reply:
yes!